src/net/sourceforge/kolmafia/session/ChoiceControl.java

```diff
@@ -4945,6 +4964,31 @@ private static String findKRAMPUSBot(final String urlString, final String respon
     return null;
   }
 
+  private static final Pattern MONSTER_ID_PATTERN = Pattern.compile("mid=(\\d+)");
+
+  private static void updateMimicMonsters(final String urlString, final int increment) {
+    var matcher = MONSTER_ID_PATTERN.matcher(urlString);
+    if (!matcher.find()) return;
+    var mid = matcher.group(1);
+
+    var updated =
+        Arrays.stream(Preferences.getString("mimicEggMonsters").split(","))
+            .map(
+                pair -> {
+                  var p = pair.split(":");
+                  var val = Integer.parseInt(p[1]);
```

Comment:
technically, this could crash if somebody sets `mimicEggMonsters` to something invalid (e.g. `342`)

I'm not sure how much we care but may as well avoid crashing in the Java?

Reply:
Yeah, I sorted this anyway in my rewrite of this function. Sorry this PR should be a draft, my mistake.